main/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,175 @@
+  disabled?: boolean
+  onChange?: (event: onChangeEvent) => void;
+  customStyle?: customStyle;
+}
+
+const Checkbox: FC<CheckboxProps> = ({
+  label,
+  value,
+  checked = false,
+  defaultChecked = false,
+  indeterminate = false,
+  disabled = false,
+  onChange,
+  customStyle,
+
+}) => {
+  const groupCheckboxContext = useContext(GroupCheckboxContext);
+
+  const isMounted = useRef(false);
+  const checkboxImg = 'https://i.imgur.com/tC6YqOV.png';
+  const [isChecked, setIsChecked] = useState(defaultChecked || checked);
+
+  const handleChange = useCallback(() => {
+    setIsChecked((prevChecked) => {
+      onChange && onChange({ checked: !prevChecked, label });
+      return !prevChecked;
+    });
+    groupCheckboxContext?.toggleOption({ label, value: value || '' });
+  }, [onChange, setIsChecked, label, groupCheckboxContext?.toggleOption]);
+
+  useEffect(() => {
+    // for only componenUpdate not in mount
+    if (isMounted.current) {
+      setIsChecked(checked);
+    } else {
+      isMounted.current = true;
+    }
+  }, [checked]);
+
+  useEffect(() => {
+    if (value && groupCheckboxContext?.registerValue) {
+      groupCheckboxContext.registerValue(value);
+    }
+  }, []);
+
+  const labelColor = disabled ? COLOR.LIGHTGRAY : customStyle?.labelColor;
+
+  return (
+    <TouchableHighlight
+      onPress={handleChange}
+      underlayColor="transparent"
+      style={{ marginHorizontal: 20, paddingBottom: 20 }}
+      disabled={disabled}>
+      <Container labelLeft={customStyle?.labelLeft}>
+        <MarkerContainer
+          boxSize={customStyle?.boxSize}
+          boxColor={disabled ? COLOR.LIGHTGRAY : customStyle?.boxColor}
+        >
+          <Marker isChecked={isChecked}>
+            {!indeterminate && isChecked && <MarkerImg
+              source={{ uri: checkboxImg }}
+            />}
+            {
+              indeterminate && <Markerindeterminate
+                boxColor={disabled ? COLOR.LIGHTGRAY : customStyle?.boxColor}
+              />
+            }
+          </Marker>
+        </MarkerContainer>
+        <Label labelColor={labelColor} labelSize={customStyle?.labelSize} labelLeft={customStyle?.labelLeft}>
+          {label}
+        </Label>
+      </Container>
+    </TouchableHighlight>
+  );
+};
+
+interface ContainerProps {
+  labelLeft?: boolean;
+}
+interface MarkerContainerProps {
+  boxSize?: number
+  boxColor?: string
+}
+
+interface LabelProps {
+  disabled?: boolean;
+  labelSize?: number;
+  labelColor?: string;
+  labelLeft?: boolean;
+}
+
+interface MarkerProps {
+  isChecked: boolean
+}
+
+interface MarkerindeterminateProps {
+  boxColor?: string
+}
+
+const COLOR: {
+  [key: string]: string;
+} = {
+  BLUE_SKY: '#1890FF',
+  LIGHTGRAY: '#E0E0E0',
+  GRAY59: '#969696',
+  BLACK: '#000000',
+};
+
+const Container = styled.View<ContainerProps>`
+ flex-direction: ${({ labelLeft }): string => labelLeft ? 'row-reverse' : 'row'};
+ justify-content: center;
+ align-items: center;
```

Comment:
We are using 2 spaces. But it looks only one space.

Reply:
Thank you for the detailed review. Could you please explain a little more?